src/OrchardCore/OrchardCore.DisplayManagement.Liquid/LiquidViewParser.cs

```diff
@@ -0,0 +1,80 @@
+            FluidValue.SetTypeMapping<ZoneHolding>(o => new ObjectValue(o));
+
+            TemplateContext.GlobalMemberAccessStrategy.Register<Shape>("*", new ShapeAccessor());
+            TemplateContext.GlobalMemberAccessStrategy.Register<ZoneHolding>("*", new ShapeAccessor());
+            TemplateContext.GlobalMemberAccessStrategy.Register<ShapeMetadata>(); TemplateContext.GlobalFilters.WithLiquidViewFilters();
+        }
+
+        public LiquidViewParser()
+        {
+            RegisterEmptyTag("render_body", RenderBodyTag.WriteToAsync);
+            RegisterParserTag("render_section", ArgumentsList, RenderSectionTag.WriteToAsync);
+            RegisterParserTag("page_title", ArgumentsList, RenderTitleSegmentsTag.WriteToAsync);
+            RegisterEmptyTag("antiforgerytoken", AntiForgeryTokenTag.WriteToAsync);
+            RegisterExpressionTag("layout", LayoutTag.WriteToAsync);
+
+            RegisterExpressionTag("shape_clear_alternates", ClearAlternatesTag.WriteToAsync);
+            RegisterParserTag("shape_add_alternates", Primary.And(Primary), AddAlternatesTag.WriteToAsync);
+            RegisterExpressionTag("shape_clear_wrappers", ClearWrappers.WriteToAsync);
+            RegisterParserTag("shape_add_wrappers", Primary.And(Primary), AddWrappersTag.WriteToAsync);
+            RegisterExpressionTag("shape_clear_classes", ClearClassesTag.WriteToAsync);
+            RegisterParserTag("shape_add_classes", Primary.And(Primary), AddClassesTag.WriteToAsync);
+            RegisterExpressionTag("shape_clear_attributes", ClearAttributesTag.WriteToAsync);
+            RegisterParserTag("shape_add_attributes", Primary.And(ArgumentsList), AddAttributesTag.WriteToAsync);
+            RegisterParserTag("shape_type", Primary.And(Primary), ShapeTypeTag.WriteToAsync);
+            RegisterParserTag("shape_display_type", Primary.And(Primary), ShapeDisplayTypeTag.WriteToAsync);
+            RegisterParserTag("shape_position", Primary.And(Primary), ShapePositionTag.WriteToAsync);
+            RegisterParserTag("shape_cache", Primary.And(ArgumentsList), ShapeCacheTag.WriteToAsync);
+            RegisterParserTag("shape_tab", Primary.And(Primary), ShapeTabTag.WriteToAsync);
+            RegisterParserTag("shape_remove_item", Primary.And(Primary), ShapeRemoveItemTag.WriteToAsync);
+            RegisterParserTag("shape_add_properties", Primary.And(ArgumentsList), ShapeAddPropertyTag.WriteToAsync);
+            RegisterParserTag("shape_remove_property", Primary.And(Primary), ShapeRemovePropertyTag.WriteToAsync);
+            RegisterParserTag("shape_pager", Primary.And(ArgumentsList), ShapePagerTag.WriteToAsync);
+
+            RegisterParserTag("httpcontext_add_items", ArgumentsList, HttpContextAddItemTag.WriteToAsync);
+            RegisterParserTag("httpcontext_remove_items", Primary, HttpContextRemoveItemTag.WriteToAsync);
+
+            RegisterParserTag("helper", ArgumentsList, FluidTagHelper.WriteArgumentsTagHelperAsync);
```

Comment:
So i imagine to move away from the tag helper bridge, and allow other modules to register tags, this class would need to be constructed from DI, and instanced, so that registration could happen, through an `IOptions` arrangement?

It would be a much nicer syntax, not to be using the helper

Reply:
Note you can already do it in this branch, since registering a tag is only about calling `_liquidViewParser.RegisterTag` (tag as in liquid tag). With this branch the tenant has its own parser instance that can be configured.

But the main issue I have with Liquid Tag -> Tag Helpers is this bridge. I'd prefer the `WriteAsync` method to just be invoking the correct code in the first place, instead of dynamically looking for a tag helper, instantiate it, map data structures, and then get back the results to write it again ... That should be doable one tag at a time, and would improve perf a lot.

Reply:
BTW I am also taking not as I look at traces on how to improve the way we pass all these services in each new template context. (IUrlHelper, DisplayHelper, ...). And I have an idea on how to fix it. And also remove the static accesses to TemplateContext such that each tenant can have their own config. Statics are bad :/